### PR TITLE
FIX: address 3rd party deprecation warnings: numba, xgboost, typing, distutils

### PR DESCRIPTION
**Pull request by connortann**
Supports #3066

Fixes or supresses various Deprecation & syntax warnings from:
- typing.io
- distutils
- sentencepiece
- keras
- xgboost
- sklearn

### CODE REVIEW

File: tests/explainers/test_tree.py
Comment (connortann) [inline tests/explainers/test_tree.py:675]: `model_output = "margin" has been renamed to model_output = "raw"`

File: shap/explainers/_deep/deep_tf.py
Comment (connortann) [inline shap/explainers/_deep/deep_tf.py:104]: Removing, as the global variable `keras` is not used in this module.

File: shap/utils/_legacy.py
Comment (thatlittleboy) [inline shap/utils/_legacy.py]: I think the default used to be 10, e.g. a very old version of sklearn 0.24: https://scikit-learn.org/0.24/modules/generated/sklearn.cluster.k_means.html#sklearn.cluster.k_means

for compatibility, it might be safer to stick to 10.

Comment (connortann) [reply shap/utils/_legacy.py]: Done

**Review by thatlittleboy — APPROVED** [review]
Approving, but I request to only merge this in after we merge in #3093. For such PRs, I think it's important to ensure the tests pass and the warnings no longer appear as advertised.

cat-standing-thankyou

**Pull request merged**
